app/controllers/users_controller.rb

```diff
@@ -0,0 +1,23 @@
+require_relative '../../lib/tunestakeoutwrapper.rb'
+
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_create_params[:user])
+    if @user.save
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def user_create_params
+    params.permit(user: [:name, :email, :password, :password_confirmation])
```

Comment:
If you use the `params.require(:user).permit(:name, :email, ...)` syntax, you won't need to hash into `user_create_params[:user]` as you did on line 10. You could just write `user_create_params` . ^_^